compute_endpoint/tests/unit/test_endpoint_unit.py

```diff
@@ -458,31 +461,29 @@ def test_always_prints_endpoint_id_to_terminal(mocker, mock_ep_data):
 
 def test_serialize_config_field_types():
     ep_config = Config()
-    ep_config._hidden_field = "123"
-
-    class S:
-        def __init__(self):
-            self.a = 1
 
-    ep_config.strategy = S()
-    ep_config.random_field = [1, 2]
+    ep_config._hidden_attr = "123"
+    ep_config.rando_attr = "howdy"
     ep_config.allowed_functions = ["a", "b", "c"]
+    ep_config.heartbeat_threshold = float("inf")
 
-    result = _serialize_config(ep_config)
+    result = serialize_config(ep_config)
```

Comment:
I see we test the hidden fields; do we want to test that fields _not_ set are also not serialized?

Reply:
All constructor parameters should be serialized, regardless whether they are set by the user. This allows for us to get a full picture of all configurable values in a given config. I believe the line right after the hidden attribute assert covers this.